packages/dma-contracts/contracts/swap/Swap.sol

```diff
@@ -127,13 +128,41 @@ contract Swap {
     amount = fromAmount.sub(feeToTransfer);
   }
 
+  function _collectFixedFee(
+    address asset,
+    uint256 fromAmount,
+    uint256 fee
+  ) internal returns (uint256 amount) {
+    amount = fromAmount.sub(fee);
+
+    if (fee > 0) {
```

Comment:
@zerotucks should I also add a check when the fee is greater than fromAmount and revert?

Reply:
Done, if fee is greater than fromAmount then we'll take whatever is available as fee. Confirmed by Chris.